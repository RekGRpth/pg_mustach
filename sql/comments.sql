\unset ECHO
\set QUIET 1
\pset format unaligned
\pset tuples_only true
\pset pager off
\set ON_ERROR_ROLLBACK 1
\set ON_ERROR_STOP true
BEGIN;
CREATE EXTENSION pg_mustach;
SELECT 1, 'Inline', replace(mustach('{}', E'12345{{! Comment Block! }}67890'), E'\n', '\n');
SELECT 2, 'Multiline', replace(mustach('{}', E'12345{{!\n  This is a\n  multi-line comment...\n}}67890\n'), E'\n', '\n');
SELECT 3, 'Standalone', replace(mustach('{}', E'Begin.\n{{! Comment Block! }}\nEnd.\n'), E'\n', '\n');
SELECT 4, 'Indented Standalone', replace(mustach('{}', E'Begin.\n  {{! Indented Comment Block! }}\nEnd.\n'), E'\n', '\n');
SELECT 5, 'Standalone Line Endings', replace(replace(mustach('{}', E'|\r\n{{! Standalone Comment }}\r\n|'), E'\r', '\r'), E'\n', '\n');
SELECT 6, 'Standalone Without Previous Line', replace(mustach('{}', E'  {{! I''m Still Standalone }}\n!'), E'\n', '\n');
SELECT 7, 'Standalone Without Newline', replace(mustach('{}', E'!\n  {{! I''m Still Standalone }}'), E'\n', '\n');
SELECT 8, 'Multiline Standalone', replace(mustach('{}', E'Begin.\n{{!\nSomething''s going on here...\n}}\nEnd.\n'), E'\n', '\n');
SELECT 9, 'Indented Multiline Standalone', replace(mustach('{}', E'Begin.\n  {{!\n    Something''s going on here...\n  }}\nEnd.\n'), E'\n', '\n');
SELECT 10, 'Indented Inline', replace(mustach('{}', E'  12 {{! 34 }}\n'), E'\n', '\n');
SELECT 11, 'Surrounding Whitespace', replace(mustach('{}', E'12345 {{! Comment Block! }} 67890'), E'\n', '\n');
ROLLBACK;
