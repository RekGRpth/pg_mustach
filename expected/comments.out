\unset ECHO
1|Inline|1234567890
2|Multiline|1234567890\n
3|Standalone|Begin.\nEnd.\n
4|Indented Standalone|Begin.\nEnd.\n
5|Standalone Line Endings||\r\n|
6|Standalone Without Previous Line|!
7|Standalone Without Newline|!\n
8|Multiline Standalone|Begin.\nEnd.\n
9|Indented Multiline Standalone|Begin.\nEnd.\n
10|Indented Inline|  12 \n
11|Surrounding Whitespace|12345  67890
